packages/nodejs/src/transmitter.ts

```diff
@@ -0,0 +1,81 @@
+import fs from "fs"
+import path from "path"
+import https from "https"
+import http from "http"
+
+import { Configuration } from "./config"
+import { URL, URLSearchParams } from "url"
+import { HashMap } from "@appsignal/types"
+
+export class Transmitter {
+  #config: Configuration
+  #url: string
+  #data: string
+
+  constructor(url: string, data = "", active = true) {
+    this.#config = new Configuration({ active })
+    this.#url = url
+    this.#data = data
+  }
+
+  public async transmit() {
+    return new Promise<HashMap<any>>((resolve, reject) => {
+      const config_data = this.#config.data
+      const params = new URLSearchParams({
+        api_key: config_data["pushApiKey"] || "",
+        name: config_data["name"] || "",
+        environment: config_data["environment"] || "",
+        hostname: config_data["hostname"] || ""
+      })
+
+      const url = new URL(`${this.#url}?${params.toString()}`)
+      const opts = this.requestOptions(url)
+      const requestModule = url.protocol == "http:" ? http : https
+
+      const request = requestModule.request(opts, (stream: any) => {
+        const responseStatus = stream.statusCode
+        stream.setEncoding("utf8")
+        let responseBody = ""
+
+        stream
+          .on("data", (chunk: any) => {
+            responseBody += chunk
+          })
+          .on("end", () => {
+            let parsedResponse
+            try {
+              parsedResponse = JSON.parse(responseBody)
+            } catch {
+              parsedResponse = {}
+            }
+
+            if (responseStatus == 200) {
+              resolve({ status: responseStatus, body: parsedResponse })
+            } else {
+              reject({ status: responseStatus, body: parsedResponse })
+            }
+          })
+      })
+
```

Comment:
If we're improving things, can we also listen to the `request.on("error", ..)` event? For when there's a problem with making the request, and not just the response. This can also be in a separate PR.

Reply:
Let's take care of it in a separate PR then

Reply:
I created an issue for it: https://github.com/appsignal/appsignal-nodejs/issues/535

Reply:
Good, let's do it after this one is merged.